doc/spec/PluginAPI.md

```diff
@@ -0,0 +1,16 @@
+# Sia-UI Plugin API Specification
+
+## Introduction
+
+This specification outlines the functionality that Sia-UI's plugin API exposes to developers.
+
+## Desired Functionality
+
+The plugin API should provide an importable javascript module that provides all of the following features:
+
+- A function which registers a plugin with Sia-UI.
+- Access to the running, configured, `sia.js` wrapper.
+- Functions to open and save files on the host.
+- Functions to control the appearance of the plugin.  This includes the ability to set the icon for the plugin and the plugin's order in the sidebar.
```

Comment:
I think it's best if we just assume a static icon location, similar to how favicons work with browsers.  The ordering will be controlled by the main UI, maybe with the ability to drag and slide plugins around the sidebar, or just with an ordering setting in the future `settings` plugin.  I've amended the spec to remove the appearance functions.